File: backend/src/main/java/com/trade/entity/Customer.java
package com.trade.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "customers", uniqueConstraints = @UniqueConstraint(columnNames = {"tenant_id", "customer_code"}))
public class Customer extends TenantAwareEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "customer_code", nullable = false, length = 50)
    private String customerCode;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(length = 50)
    private String contact;

    @Column(length = 20)
    private String phone;

    @Column(length = 200)
    private String address;

    @Enumerated(EnumType.STRING)
    private CustomerStatus status = CustomerStatus.ACTIVE;

    @CreationTimestamp
    private LocalDateTime createTime;

    @UpdateTimestamp
    private LocalDateTime updateTime;

    public enum CustomerStatus {
        ACTIVE, INACTIVE
    }
}

File: webchat_app/utils/customer.js
export const CUSTOMER_STATUS = {
  ACTIVE: '启用',
  INACTIVE: '停用',
};

export function customerStatusTheme(status) {
  return status === 'ACTIVE' ? 'success' : 'default';
}

export function mapCustomer(row) {
  return {
    id: row.id,
    name: row.name || '—',
    contact: row.contact || '—',
    phone: row.phone || '—',
    address: row.address || '',
    status: row.status || 'ACTIVE',
    statusText: CUSTOMER_STATUS[row.status] || row.status || '—',
    statusTheme: customerStatusTheme(row.status),
  };
}
